lib_expense/src/main/java/com/example/expenselogger/lib_expense/domain/impl/ExpenseContractImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.expenselogger.lib_expense.domain.impl
+
+import com.example.expenselogger.cache.repository.ExpenseRepository
+import com.example.expenselogger.lib_expense.domain.contract.ExpenseContract
+import com.example.expenselogger.lib_expense.domain.mapper.ExpenseModelMapper
+import com.example.expenselogger.lib_expense.domain.model.Expense
+import javax.inject.Inject
+
+internal class ExpenseContractImpl @Inject constructor(
+    private val expenseRepository: ExpenseRepository,
+    private val expenseModelMapper: ExpenseModelMapper
+) : ExpenseContract {
+    override suspend fun saveExpense(expense: Expense) {
+        if (expense.id <= 0L) {
```

Comment:
Oh what's the idea behind this check

Reply:
I added a `createExpense` in [Expense](https://github.com/larmie56/ExpenseLogger/blob/79371823efe21ff9e78fcaec7520940d7f7c337c/lib_expense/src/main/java/com/example/expenselogger/lib_expense/domain/model/Expense.kt); so the expense object can always use 0 as the id of new insertions, but if it's not a new insertion (i.e. update), it'll already have an id that is > 0. So I felt that's a way to go about handling insertion and update

Reply:
Okay. Not bad. But there's no need for that much logic when you can just need an additional function to handle updates in the expense contract.
Initially I thought we could use one function for both but that's not possible with auto Gen primary key

Reply:
Yeaa, I considered that, but I was thinking maybe I should let the contract stay the way it is, and work with what's there. I will change it to add the extra function. Thank you!

Reply:
Always go for the neater approach

Reply:
Upon adding the extra function now, it turns out that it is a little bit more involved. We have to had an extra usecase too (SaveExpense usecase will not be sufficient anymore). I'm already doing that now, just updating you on it

Reply:
> Always go for the neater approach, even if it will take more effort

Thank you very much for point it out, I'll always do that moving forward